sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/security/handler/ObscuringErrorHandlerTest.java

```diff
@@ -0,0 +1,123 @@
+
+  @Test
+  void shouldTransformToStandardErrors() {
+    var headers = new HttpHeaders();
+    headers.setContentType(MediaType.APPLICATION_JSON);
+    ResponseEntity<ApiError> exchange =
+        client.exchange(
+            getServerBaseUrl() + "/api/errorEntity",
+            HttpMethod.GET,
+            new HttpEntity<>(headers),
+            ApiError.class);
+    assertThat(exchange)
+        .isNotNull()
+        .extracting(ResponseEntity::getBody)
+        .extracting(ApiError::getTitle)
+        .isEqualTo("HTTP Error 500 occurred.");
+  }
+
+  @Test
+  void shouldNotTransformApiErrors() {
+    var headers = new HttpHeaders();
+    headers.setContentType(MediaType.APPLICATION_JSON);
+    ResponseEntity<ApiError> exchange =
+        client.exchange(
+            getServerBaseUrl() + "/api/apiError",
+            HttpMethod.GET,
+            new HttpEntity<>(headers),
+            ApiError.class);
+    assertThat(exchange)
+        .isNotNull()
+        .extracting(ResponseEntity::getStatusCode, r -> r.getBody().getTitle())
+        .containsExactly(HttpStatus.NOT_IMPLEMENTED, "This method is not implemented yet.");
+  }
+
+  @Test
+  void shouldMapExceptions() {
+    var headers = new HttpHeaders();
+    headers.setContentType(MediaType.APPLICATION_JSON);
+    ResponseEntity<ApiError> exchange =
+        client.exchange(
+            getServerBaseUrl() + "/api/forcedError",
+            HttpMethod.GET,
+            new HttpEntity<>(headers),
+            ApiError.class);
+    assertThat(exchange)
+        .isNotNull()
+        .extracting(ResponseEntity::getStatusCode, r -> r.getBody().getTitle())
+        .containsExactly(HttpStatus.INTERNAL_SERVER_ERROR, "An exception occurred.");
+  }
+
+  @Test
+  void shouldGetResponseEntityWithBody() {
+    var headers = new HttpHeaders();
+    headers.setContentType(MediaType.APPLICATION_JSON);
+    ResponseEntity<TestResource> exchange =
+        client.exchange(
+            getServerBaseUrl() + "/api/response",
+            HttpMethod.GET,
+            new HttpEntity<>(headers),
+            TestResource.class);
+    assertThat(exchange)
+        .isNotNull()
+        .extracting(ResponseEntity::getStatusCode, r -> r.getBody().getValue())
+        .containsExactly(HttpStatus.CREATED, "This will not be altered.");
+  }
+
+  @Test
+  void shouldGetResource() {
+    var headers = new HttpHeaders();
+    headers.setContentType(MediaType.APPLICATION_JSON);
+    ResponseEntity<TestResource> exchange =
+        client.exchange(
+            getServerBaseUrl() + "/api/resource",
+            HttpMethod.GET,
+            new HttpEntity<>(headers),
+            TestResource.class);
+    assertThat(exchange)
+        .isNotNull()
+        .extracting(ResponseEntity::getStatusCode, r -> r.getBody().getValue())
+        .containsExactly(HttpStatus.OK, "This will not be altered.");
+  }
+
+  String getServerBaseUrl() {
+    return String.format("http://localhost:%s", port);
+  }
```

Comment:
Strange that Spring does not offer a more elegant way to retrieve the base URL. [Official documentation](https://spring.io/guides/gs/testing-web/) contains the same example.